python/cudf/cudf/_libxx/binaryop.pyx

```diff
@@ -150,6 +153,21 @@ cdef binaryop_s_v(Scalar lhs, Column rhs,
     return Column.from_unique_ptr(move(c_result))
 
 
+def handle_null_for_string_column(Column input_col, op):
+    if op in (
+        BinaryOperation['EQ'], BinaryOperation['LT'],
```

Comment:
I think `op` here should just be a string (`eq`, `lt`, etc.,). The conversion to a `BinaryOperation` is extraneous.

Something that may help here: if `x` is a `BinaryOperation['SUM']` for example, then `x.name.lower() == 'sum'`.